hypershift-operator/main.go

```diff
@@ -438,3 +444,127 @@ func run(ctx context.Context, opts *StartOptions, log logr.Logger) error {
 	log.Info("starting manager")
 	return mgr.Start(ctx)
 }
+
+func NewInitCommand() *cobra.Command {
+	ctrl.SetLogger(zap.New(zap.UseDevMode(true), zap.JSONEncoder(func(o *zapcore.EncoderConfig) {
+		o.EncodeTime = zapcore.RFC3339TimeEncoder
+	})))
+
+	cmd := &cobra.Command{
+		Use:   "init",
+		Short: "Initializes the environment for the Hypershift Operator",
+	}
+
+	cmd.Run = func(cmd *cobra.Command, args []string) {
+		ctx, cancel := context.WithCancel(ctrl.SetupSignalHandler())
+		defer cancel()
+
+		if err := runInit(ctx, ctrl.Log.WithName("init")); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	}
+
+	return cmd
+}
+
+func runInit(ctx context.Context, log logr.Logger) error {
+	log.Info("Initializing environment for Hypershift Operator")
+	client, err := util.GetClient()
+	if err != nil {
+		return err
+	}
+	config, err := ctrl.GetConfig()
+	if err != nil {
+		return err
+	}
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(config)
+	if err != nil {
+		return fmt.Errorf("unable to get discovery client: %w", err)
+	}
+	mgmtClusterCaps, err := capabilities.DetectManagementClusterCapabilities(discoveryClient)
+	if err != nil {
+		return fmt.Errorf("unable to detect management cluster capabilities: %w", err)
+	}
+
+	trustedCABundle := new(bytes.Buffer)
+	var ocmTrustedCABundle []byte
+	if _, err := os.Stat("/var/run/ca-trust/tls-ca-bundle.pem"); err == nil {
+		ocmTrustedCABundle, err = ioutil.ReadFile("/var/run/ca-trust/tls-ca-bundle.pem")
+		if err != nil {
+			return fmt.Errorf("unable to read ocm CA trust bundle file: %w", err)
+		}
+	} else if err != nil && errors.Is(err, os.ErrNotExist) {
+		ocmTrustedCABundle, err = ioutil.ReadFile("/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem")
+		if err != nil {
+			return fmt.Errorf("unable to read ocm CA trust bundle file: %w", err)
+		}
+	} else {
+		return err
+	}
+
+	if _, err := trustedCABundle.Write(ocmTrustedCABundle); err != nil {
+		return fmt.Errorf("unable to write ocm CA trust bundle to buffer: %w", err)
+	}
+
+	if mgmtClusterCaps.Has(capabilities.CapabilityImage) {
+		// Adds user-specified image registry CA bundle to the full CA trust bundle if it exists
+		imageRegistryCABundle, err := getImageRegistryCABundle(ctx, client)
+		if err != nil {
+			return fmt.Errorf("unable to get CA bundle for image mirror registries: %w", err)
+		}
+		if imageRegistryCABundle != nil {
+			if _, err := trustedCABundle.Write(imageRegistryCABundle.Bytes()); err != nil {
+				return fmt.Errorf("unable to write image registry CA to buffer: %w", err)
+			}
+		}
+	}
+
+	if err := ioutil.WriteFile("/trust-bundle/tls-ca-bundle.pem", trustedCABundle.Bytes(), 0644); err != nil {
```

Comment:
is /trust-bundle  path arbitrary or is there something we can document for where is coming from?

Reply:
This is the path mounted to the init container https://github.com/openshift/hypershift/pull/2713/files#diff-4edc0a874ce10f9d4f77594c9ff489c3275cb2adc788826a9ea2f48a64c336efR624

The path mounted is arbitrary, should this be documented that this is the expected path somehow (as a const?) or should I choose a different path that makes more sense?